### PR TITLE
Ataymano/memory leaks fixes

### PR DESCRIPTION
**Pull request by ataymano**
Unit tests memory leaks fixes

### CODE REVIEW

File: test/unit_test/ccb_parser_test.cc [hunk 1]
```diff
@@ -9,91 +9,127 @@
...
+  p.parse_name.delete_v();
 }
 
 BOOST_AUTO_TEST_CASE(ccb_cache_label)
 {
   io_buf io;
-  io.init();
-  io.space.resize(1000);
-  io.space.end() = io.space.begin() + 1000;
+  //io.init();      TODO: figure out and fix leak caused by double init()
```
Comment (jackgerrits) [inline test/unit_test/ccb_parser_test.cc]: This line can just be removed

**Comment by jackgerrits** [comment]
Thanks for these fixes @ataymano!

In my mind this PR just points out the critical need for RAII types in VW... The fact we can leak so easily by forgetting to call a function is an issue. Destructors fix this automatically

**Comment by ataymano** [comment]
> Thanks for these fixes @ataymano!
> 
> In my mind this PR just points out the critical need for RAII types in VW... The fact we can leak so easily by forgetting to call a function is an issue. Destructors fix this automatically

Oh, I missed your PR about destructors. Will cancel this one then.

**Comment by jackgerrits** [comment]
No no we still need this, I also am not sure if I will be able to fix that PR anytime soon.

File: test/unit_test/ccb_parser_test.cc (hunk 1 above)
Comment (ataymano) [reply test/unit_test/ccb_parser_test.cc]: I left it for TODO comment

**Comment by jackgerrits** [comment]
Thanks @ataymano!

**Pull request merged**